src/services/socket.js

```diff
@@ -0,0 +1,61 @@
+import {
+  connected,
+  authenticateSocket,
+  logoutSocket,
+  clearCookie,
+  handleResponse
+} from '../utils';
+
+class SocketService {
+  constructor (app) {
+    this.app = app;
+  }
+
+  create (data, params) {
+    const app = this.app;
+
+    return app.authentication.getJWT().then(token => {
+      data.accessToken = token;
+      console.log('token', token);
+
+      return connected(app).then(socket => {
+        const method = app.io ? 'emit' : 'send';
+        console.log('method', method);
+        return authenticateSocket(data, socket, method)
```

Comment:
In latest those are gone. I guess we can bring them back but the better way is to pass a `callback` to the `authenticate` event which will be called back with the error or an object with the access token.